src/wallet/rpcdisclosure.cpp

```diff
@@ -0,0 +1,296 @@
+// Copyright (c) 2017 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "base58.h"
+#include "rpcserver.h"
+#include "init.h"
+#include "main.h"
+#include "script/script.h"
+#include "script/standard.h"
+#include "sync.h"
+#include "util.h"
+#include "utiltime.h"
+#include "wallet.h"
+
+#include <fstream>
+#include <stdint.h>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/date_time/posix_time/posix_time.hpp>
+
+#include <univalue.h>
+
+#include "paymentdisclosure.h"
+#include "paymentdisclosuredb.h"
+
+#include "zcash/Note.hpp"
+#include "zcash/NoteEncryption.hpp"
+
+using namespace std;
+using namespace libzcash;
+
+void EnsureWalletIsUnlocked();
+bool EnsureWalletIsAvailable(bool avoidException);
```

Comment:
Add a comment explaining where these are implemented (and maybe why they are re-declared here).

Reply:
Ok